neqo-transport/src/connection/tests/handshake.rs

```diff
@@ -208,7 +208,7 @@ fn dup_server_flight1() {
     assert!(out.as_dgram_ref().is_some());
     qdebug!("Output={:0x?}", out.as_dgram_ref());
 
-    assert_eq!(4, client.stats().packets_rx);
+    assert_eq!(3, client.stats().packets_rx);
```

Comment:
Most of the changes below simply revert changes made in https://github.com/mozilla/neqo/pull/2389/.

https://github.com/mozilla/neqo/pull/2389/ adjusted the tests of larger flight sizes due to MLKEM.

No longer doing ECN during the handshake replaces all ACK-ECN with ACK, thus oftentimes reducing the flight sizes again.